tsl/src/compression/compression.c

```diff
@@ -631,21 +856,46 @@ row_compressor_init(RowCompressor *row_compressor, TupleDesc uncompressed_tuple_
 				.min_metadata_attr_offset = segment_min_attr_offset,
 				.max_metadata_attr_offset = segment_max_attr_offset,
 				.min_max_metadata_builder = segment_min_max_builder,
+				.segmentby_column_index = -1,
 			};
 		}
 		else
 		{
+			is_segment_by = true;
 			if (column_attr->atttypid != compressed_column_attr->atttypid)
 				elog(ERROR,
 					 "expected segment by column \"%s\" to be same type as uncompressed column",
 					 compression_info->attname.data);
 			*column = (PerColumn){
 				.segment_info = segment_info_new(column_attr),
+				.segmentby_column_index = compression_info->segmentby_column_index,
 				.min_metadata_attr_offset = -1,
 				.max_metadata_attr_offset = -1,
 			};
 		}
 	}
+
+	row_compressor->index_oid =
+		get_compressed_chunk_index(compressed_table,
+								   row_compressor->uncompressed_col_to_compressed_col,
+								   row_compressor->per_column,
+								   row_compressor->n_input_columns);
+
+	/* We expect the appropriate index to exist for segment by compression
+	 * if the table is not empty. If it is empty, this is not a problem
+	 * since we are not merging chunks.
+	 */
```

Comment:
Hmm why is index a requirement here, we can fall back to sequential scan if no index is present

Reply:
Fixed so its not a requirement.